### PR TITLE
clarify maximum number of buckets

### PR DESCRIPTION
**Pull request by harshavardhana**
## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description
clarify the maximum number of buckets

## Motivation and Context
just keep it more educational and pragmatic

## How to test this PR?
Nothing should change for CI/CD

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [x] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: docs/minio-limits.md [hunk 1]
```diff
@@ -34,6 +34,8 @@ For optimal production setup MinIO recommends Linux kernel version 4.x and later
 | Maximum length for '/' separated object name segment                            | 255                                                                             |
 | Maximum number of versions per object                                           | 10000 (can be configured to higher values but we do not recommend beyond 10000) |
 
+> NOTE: Number of buckets in any cluster is a function of the hardware limits and what kind of workload it would scale for. We advise that you consult us directly via https://min.io/pricing for better sizing guidelines for your use-case.
```
Comment (ravindk89) [inline docs/minio-limits.md]:
```suggestion
> NOTE:  While MinIO does not implement an upper boundary on buckets, your cluster's hardware has natural limits that depend on the workload and it's scaling patterns. We strongly recommend [MinIO SUBNET](https://min.io/pricing) for architecture and sizing guidance for your production use case.
```

Comment (ravindk89) [reply docs/minio-limits.md]: Slight rewrite.

**Review by ravindk89 — APPROVED** [review]
LGTM or one nit

**Review by ravindk89 — APPROVED** [review]

File: docs/minio-limits.md [hunk 2]
```diff
@@ -34,6 +34,8 @@ For optimal production setup MinIO recommends Linux kernel version 4.x and later
 | Maximum length for '/' separated object name segment                            | 255                                                                             |
 | Maximum number of versions per object                                           | 10000 (can be configured to higher values but we do not recommend beyond 10000) |
 
+> NOTE:  While MinIO does not implement an upper boundary on buckets, your cluster's hardware has natural limits that depend on the workload and it's scaling patterns. We strongly recommend [MinIO SUBNET](https://min.io/pricing) for architecture and sizing guidance for your production use case.
```
Comment (djwfyi) [inline docs/minio-limits.md]:
```suggestion
> NOTE:  While MinIO does not implement an upper boundary on buckets, your cluster's hardware has natural limits that depend on the workload and its scaling patterns. We strongly recommend [MinIO SUBNET](https://min.io/pricing) for architecture and sizing guidance for your production use case.
```

**Review by djwfyi — CHANGES_REQUESTED** [review]
One tiny grammar thing.

**Review by djwfyi — APPROVED** [review]

**Pull request merged**